apps/opik-backend/src/main/java/com/comet/opik/domain/DatasetService.java

```diff
@@ -143,7 +145,8 @@ public Dataset save(@NonNull Dataset dataset) {
             } catch (UnableToExecuteStatementException e) {
                 if (e.getCause() instanceof SQLIntegrityConstraintViolationException) {
                     log.info(DATASET_ALREADY_EXISTS);
-                    throw new EntityAlreadyExistsException(new ErrorMessage(List.of(DATASET_ALREADY_EXISTS)));
+                    throw new EntityAlreadyExistsException(
+                            new ErrorMessage(CONFLICT.getStatusCode(), DATASET_ALREADY_EXISTS));
```

Comment:
EntityAlreadyExistsException, already deals with status code, believe that the status code isn't needed here

Let's go with Encapsulation & DRY principles